math/dec.go

```diff
@@ -61,21 +70,41 @@ func NewDecFromString(s string) (Dec, error) {
 	}
 }
 
+// NewDecFromInt64 converts an int64 to a Dec type.
+// This function is useful for creating Dec values from integer literals or variables,
+// ensuring they can be used in high-precision arithmetic operations defined for Dec types.
+//
+// Example:
+// - NewDecFromInt64(123) returns a Dec representing the value 123.
 func NewDecFromInt64(x int64) Dec {
 	var res Dec
 	res.dec.SetInt64(x)
 	return res
 }
 
-// NewDecWithPrec returns a decimal with a value of coeff * 10^exp precision.
+// NewDecWithPrec creates a Dec from a coefficient and exponent, calculated as coeff * 10^exp.
+// Useful for precise decimal representations.
+//
+// Example:
+// - NewDecWithPrec(123, -2) -> Dec representing 1.23.
 func NewDecWithPrec(coeff int64, exp int32) Dec {
 	var res Dec
 	res.dec.SetFinite(coeff, exp)
 	return res
 }
 
-// Add returns a new Dec with value `x+y` without mutating any argument and error if
-// there is an overflow.
+// Add returns a new Dec representing the sum of `x` and `y` using decimal128 precision.
+// This function ensures that no arguments are mutated during the operation and checks for overflow conditions.
+// If an overflow occurs, an error is returned.
+//
+// Key error scenarios:
+// - Overflow if the sum exceeds the maximum representable value within decimal128 precision.
```

Comment:
`decimal128 precision` is not correct. The precision is much higher as long as the max exponent is not exceeded.  `apd.BaseContext` is used and not `dec128Context`.
It would be good to explain this with some examples. It is not intuitive that `1e100000 + 0 -> err` for example